projects/can_transceiver/src/can_transceiver_ros_intf.cpp

```diff
@@ -44,12 +60,117 @@ class CanTransceiverIntf : public rclcpp::Node
+        timer_ = this->create_wall_timer(
+          std::chrono::milliseconds(PLACEHOLDER_MS), std::bind(&CanSimIntf::timer_callback, this));
+    }
+
+private:
+    // Subscriber
+    void gps_callback(const custom_interfaces::msg::GPS::SharedPtr msg) const {}
+    void wind_callback(const custom_interfaces::msg::WindSensors::SharedPtr msg) const {}
+
+    // Publisher
+    void timer_callback()
+    {
+        // ** GPS **
+        custom_interfaces::msg::GPS gps_data;
+        gps_data.heading.heading   = PLACEHOLDER_VALUE;
+        gps_data.speed.speed       = PLACEHOLDER_VALUE;
+        gps_data.lat_lon.latitude  = PLACEHOLDER_VALUE;
+        gps_data.lat_lon.longitude = PLACEHOLDER_VALUE;
+        // Publishes msg
+        pub_gps_->publish(gps_data);
+
+        // ** Filtered Wind Sensor **
+        custom_interfaces::msg::WindSensor filtered_wind_sensor_data;
+        filtered_wind_sensor_data.speed.speed = PLACEHOLDER_VALUE;
+        filtered_wind_sensor_data.direction   = PLACEHOLDER_VALUE;
+        // Publishes msg
+        pub_filtered_wind_sensor_->publish(filtered_wind_sensor_data);
+
+        // ** Wind Sensors **
+        custom_interfaces::msg::WindSensors wind_sensors_data;
+        for (auto & sensor : wind_sensors_data.wind_sensors) {
+            sensor.direction   = PLACEHOLDER_VALUE;
+            sensor.speed.speed = PLACEHOLDER_VALUE;
+        }
+        // Publishes msg
+        pub_wind_sensors_->publish(wind_sensors_data);
+
+        // ** Batteries **
+        custom_interfaces::msg::Batteries batteries_data;
+        for (auto & sensor : batteries_data.batteries) {
+            sensor.voltage = PLACEHOLDER_VALUE;
+            sensor.current = PLACEHOLDER_VALUE;
+        }
+        // Publishes msg
+        pub_batteries_->publish(batteries_data);
+
+        // ** Data Sensors **
+        custom_interfaces::msg::HelperGenericSensor helper_gen_sensor_data;
+        custom_interfaces::msg::GenericSensors      gen_sensor_data;
+        helper_gen_sensor_data.id   = 0x0;  //uint8
+        helper_gen_sensor_data.data = 0x0;  //uint64
+        // Publishes msg
+        gen_sensor_data.generic_sensors.push_back(helper_gen_sensor_data);
+        pub_data_sensors_->publish(gen_sensor_data);
+    }
+
+    // Subscriber Field Declarations
+    rclcpp::Subscription<custom_interfaces::msg::GPS>::SharedPtr         sub_gps_;
+    rclcpp::Subscription<custom_interfaces::msg::WindSensors>::SharedPtr sub_mock_wind_sensors_;
+    // Publisher Field Declarations
+    rclcpp::Publisher<custom_interfaces::msg::GPS>::SharedPtr            pub_gps_;
+    rclcpp::Publisher<custom_interfaces::msg::WindSensor>::SharedPtr     pub_filtered_wind_sensor_;
+    rclcpp::Publisher<custom_interfaces::msg::WindSensors>::SharedPtr    pub_wind_sensors_;
+    rclcpp::Publisher<custom_interfaces::msg::Batteries>::SharedPtr      pub_batteries_;
+    rclcpp::Publisher<custom_interfaces::msg::GenericSensors>::SharedPtr pub_data_sensors_;
```

Comment:
CanTransceiverIntf should publish these sensors to the regular ROS network, not the CanSimIntf. CanSimIntf should only interact with the simulator side of the ROS network.

Reply:
Split into two nodes